src/wallet/wallet.cpp

```diff
@@ -3205,6 +3205,9 @@ void CWallet::SetupDescriptorScriptPubKeyMans()
                 FlatSigningProvider keys;
                 std::string dummy_error;
                 std::unique_ptr<Descriptor> desc = Parse(desc_str, keys, dummy_error, false);
+                if (desc == NULL) {
+                    throw std::runtime_error(std::string(__func__) + ": Invalid descriptor");
```

Comment:
Adding `desc_str` to the result would be useful, so the user doesn't have to manually call HWI and doesn't have to guess which of the descriptors failed.

You could rename `dummy_error` to `desc_error` and include that in the message too.

Reply:
done in https://github.com/bitcoin/bitcoin/pull/23628/commits/a5f53c36942bf82c056380df6c15adf0b12190f2